pkg/cmd/admin/policy/subject_review.go

```diff
@@ -0,0 +1,200 @@
+	o := &subjectReviewOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Long:    subjectReviewLong,
+		Short:   "Check whether a user or a ServiceAccount can create a Pod.",
+		Example: fmt.Sprintf(subjectReviewExamples, fullName, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, out))
+			kcmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
+	kcmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
+	cmd.Flags().StringVarP(&o.User, "user", "u", o.User, "Review will be performed on behalf of this user")
+	cmd.Flags().StringSliceVarP(&o.Groups, "groups", "g", o.Groups, "Comma separated, list of groups. Review will be performed on behalf of these groups")
+	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "z", o.ServiceAccount, "service account in the current namespace to use as a user")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *subjectReviewOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer) error {
+	if len(o.User) > 0 && len(o.ServiceAccount) > 0 {
+		return fmt.Errorf("--user and --serviceaccount are mutually exclusive")
+	}
+	var err error
+	o.namespace, o.enforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.subjectReviewClient = oclient
+	o.selfSubjectReviewClient = oclient
+	o.mapper, o.typer = f.Object()
+	o.RESTClientFactory = f.ClientForMapping
+
+	if len(kcmdutil.GetFlagString(cmd, "output")) != 0 {
+		clientConfig, err := f.ClientConfig()
+		if err != nil {
+			return err
+		}
+		version, err := kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+		if err != nil {
+			return err
+		}
+		p, _, err := kcmdutil.PrinterForCommand(cmd)
+		if err != nil {
+			return err
+		}
+		o.printer = kubectl.NewVersionedPrinter(p, kapi.Scheme, version)
+	} else {
+		o.printer = describe.NewHumanReadablePrinter(kubectl.PrintOptions{NoHeaders: false})
+	}
+
+	o.out = out
+	o.approximatePodTemplateForObject = f.ApproximatePodTemplateForObject
+	return nil
+}
+
+func (o *subjectReviewOptions) Run(args []string) error {
+	userOrSA := o.User
+	if len(o.ServiceAccount) > 0 {
+		userOrSA = o.ServiceAccount
+	}
+	subjectReviews := []*securityapi.PodSecurityPolicySubjectReview{}
+	selfSubjectReviews := []*securityapi.PodSecurityPolicySelfSubjectReview{}
+	allErrs := []error{}
+
+	infos, err := resource.NewBuilder(o.mapper, o.typer, resource.ClientMapperFunc(o.RESTClientFactory), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(o.namespace).
+		FilenameParam(o.enforceNamespace, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Flatten().
+		Do().
+		Infos()
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case len(userOrSA) > 0 || len(o.Groups) > 0:
```

Comment:
I doubt you'll be doing both at the same time, either you do specify user and/or group or you don't and invoke self check. I'd split that into two methods one for each check. The code for the two can be shared start from line 148 until 154. The only difference will be setting `.spec.User` and/or `.spec.Groups` fields and the actual client calls. Printing should be shared as well.